### PR TITLE
Get node IP from GetNodes() instead of net.Lookup()

### PR DESCRIPTION
**Pull request by pravisankar**
This will avoid cases where node IP resolution on
master and node differs for various reasons (stale dns cache, etc.)

### CODE REVIEW

**Comment by pravisankar** [comment]
@rajatchopra @dcbw @danwinship PTAL

File: ovssubnet/registry/registry.go [hunk 1]
```diff
@@ -142,28 +132,35 @@ func (sub *EtcdSubnetRegistry) InitNodes() error {
 	return err
 }
 
-func (sub *EtcdSubnetRegistry) GetNodes() (*[]string, error) {
+func (sub *EtcdSubnetRegistry) GetNodes() (*[]string, *[]string, error) {
```
Comment (rajatchopra) [inline ovssubnet/registry/registry.go]: Let's return node objects (create new if needed). Two arrays of strings seems odd.

File: ovssubnet/registry/registry.go [hunk 2]
```diff
@@ -42,24 +42,14 @@ func newNodeEvent(action, key, value string) *api.NodeEvent {
...
-					}
-				}
-			}
+			nodeEvent.NodeIP = nodeIP
+			return nodeEvent
 		}
 	}
 
-	fmt.Printf("Error decoding node event: nil key (%s,%s,%s).\n", action, key, value)
+	fmt.Errorf("Error decoding node event: nil key (%s,%s,%s).\n", action, key, value)
```
Comment (danwinship) [inline ovssubnet/registry/registry.go]: fmt.Errorf() returns an error object and doesn't print anything. I think you mean "fmt.Fprintf(os.Stderr, ...)", but we don't do that in most other places, so if we're going to start fixing that, it should probably be globally, in a separate PR

Comment (pravisankar) [inline ovssubnet/registry/registry.go]: There is a lot of inconsistency in registry.go for reporting errors. I'll handle this in a separate PR.

**Comment by pravisankar** [comment]
@rajatchopra @danwinship Updated, please review/merge

File: ovssubnet/common.go [hunk 3]
```diff
@@ -446,3 +413,24 @@ func (oc *OvsController) Stop() {
 	close(oc.sig)
 	//oc.sig <- struct{}{}
 }
+
+func GetNodeIP(nodeName string) (string, error) {
```
Comment (rajatchopra) [inline ovssubnet/common.go]: Should this be a public function? getNodeIP was good, right?

**Comment by dcbw** [comment]
LGTM now

**Pull request merged**